nibabel/cmdline/diff.py

```diff
@@ -0,0 +1,81 @@
+#!python
+# emacs: -*- mode: python-mode; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the NiBabel package for the
+#   copyright and license terms.
+#
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""
+Output a summary table for neuroimaging files (resolution, dimensionality, etc.)
+"""
+from __future__ import division, print_function, absolute_import
+
+import sys
+from optparse import OptionParser, Option
+
+import numpy as np
+
+import nibabel as nib
+import nibabel.cmdline.utils
+from nibabel.cmdline.utils import _err, verbose, table2string, ap, safe_get
+import fileinput
+
+__author__ = 'Yaroslav Halchenko & Christopher Cheng'
+__copyright__ = 'Copyright (c) 2017 NiBabel contributors'
+__license__ = 'MIT'
+
+
+def get_opt_parser():
+    # use module docstring for help output
+    p = OptionParser(
+        usage="%s [OPTIONS] [FILE ...]\n\n" % sys.argv[0] + __doc__,
+        version="%prog " + nib.__version__)
+
+    p.add_options([
+        Option("-v", "--verbose", action="count",
+               dest="verbose", default=0,
+               help="Make more noise.  Could be specified multiple times"),
+
+        Option("-H", "--header-fields",
+               dest="header_fields", default='',
+               help="Header fields (comma separated) to be printed as well (if present)"),
+    ])
+
+    return p
+
+
+def diff_dicts(key, compare1, compare2):
```

Comment:
I do not see anything in this function comparing actually dictionaries -- it is just a comparison of two values for a key and then spitting out a dictionary with that key.  so it could/should be just `diff_values(value1, value2)` and `diff_dicts` should be what it used to be -- get entire dictionaries, and then internally possibly use this `diff_values` to compare values (what you now do within `main()`)

Reply:
I just kept calling it diff_dicts because it compared differences and spit out a dict :P But with the header I don't think comparisons of dicts are necessary (but I'd love to be persuaded otherwise). Otherwise noted! Will get cracking on it :-)